pkg/schema/searchComplete_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright Contributors to the Open Cluster Management project
+package schema
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stolostron/search-v2-api/graph/model"
+)
+
+func Test_SearchComplete_Query(t *testing.T) {
+	// Create a SearchCompleteResolver instance with a mock connection pool.
+	prop1 := "kind"
+	searchInput := &model.SearchInput{}
+	resolver, mockPool := newMockSearchComplete(t, searchInput, prop1)
+	val1 := "Template"
+	val2 := "ReplicaSet"
+	expectedProps := []*string{&val1, &val2}
+
+	// Mock the database queries.
+	mockRows := newMockRows("../resolver/mocks/mock.json")
+	// Mock the database query
+	mockPool.EXPECT().Query(gomock.Any(),
+		gomock.Eq(`SELECT DISTINCT "data"->>'kind' FROM "search"."resources" WHERE ("data"->>'kind' IS NOT NULL) ORDER BY "data"->>'kind' ASC LIMIT 10000`),
+		gomock.Eq([]interface{}{})).Return(mockRows, nil)
+
+	// Execute function
+	result, err := resolver.autoComplete(context.TODO())
+	fmt.Println("err from autoComplete: ", err)
+	// Verify response
+	if !string_array_equal(result, expectedProps) {
+		t.Errorf("Incorrect Result() expected [%v] got [%v]", expectedProps, result)
+	}
+}
+
+func Test_SearchCompleteNoProp_Query(t *testing.T) {
+	// Create a SearchCompleteResolver instance with a mock connection pool.
+	prop1 := ""
+	searchInput := &model.SearchInput{}
+	resolver, mockPool := newMockSearchComplete(t, searchInput, prop1)
+	val1 := "Template"
+	val2 := "ReplicaSet"
+	expectedProps := []*string{&val1, &val2}
+
+	// Mock the database queries.
+	mockRows := newMockRows("../resolver/mocks/mock.json")
+	// Mock the database query
+	mockPool.EXPECT().Query(gomock.Any(),
+		gomock.Eq(""),
+		gomock.Eq([]interface{}{})).Return(mockRows, nil)
+
+	// Execute function
+	result, _ := resolver.autoComplete(context.TODO())
+
+	// Verify response
+	if !string_array_equal(result, expectedProps) {
+		t.Errorf("Incorrect Result() expected [%v] got [%v]", expectedProps, result)
+	}
+}
+
+func string_array_equal(result, expected []*string) bool { //, expected []interface{}) bool {
+	for i, exp := range expected {
+		if *result[i] != *exp {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
This is a nice function to reuse across tests. I'm just adding a similar one to the indexer [in this PR](https://github.com/stolostron/search-indexer/pull/31/files#diff-ee17c67bdc69ad2fdca5d64f2a447c755f5efbf88df225b38aa29826c2bd8dedR19-R24)

I want to suggest a few changes to keep consistency.
```suggestion
func AssertStringArrayEqual(t *testing.T, result, expected []*string, message string) {
	for i, exp := range expected {
		if *result[i] != *exp {
			t.Errorf("%s expected [%v] got [%v]", message, expected, result)
			return
		}
	}
}
```